OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -71,7 +61,13 @@ public AmmoPool(Actor self, AmmoPoolInfo info)
 			else
 				CurrentAmmo = Info.Ammo;
 
-			RemainingTicks = Info.SelfReloadTicks;
+			reload = self.TraitsImplementing<ReloadAmmo>()
+				.Where(a => a.Info.ReloadAmmoPools.Contains(Info.Name))
+				.FirstOrDefault();
```

Comment:
This line here is the reason why the minelayer doesn't reload. In the minelayer's yaml, the `ReloadsAmmo` trait comes after `AmmoPool`. You need to implemented `INotifyCreated` move this line into the `Created()` method.